### ISSUE TITLE
Crashes the server as it causes it to continually compact the database for combats

### ISSUE THREAD

**Issue by strongpauly** [vtt-lair/foundry-group-initiative#7]
Not entirely sure of the steps to reproduce, or if it was solely this modules fault, but just ran a session with a grouped initiative combat that caused several server side errors, which eventually caused the server to need to be restarted as it seemingly ran into an infinite loop compacting the database for combat.

Server side errors seen were: 
```
Error: The key MGTKlyoHKRynj12l does not exist in the EmbeddedCollection Collection
    at Map.get (file:///home/ubuntu/app/resources/app/common/utils/collection.mjs:106:13)
    at ServerDatabaseBackend._updateEmbeddedDocuments (file:///home/ubuntu/app/resources/app/dist/database/backend/server-backend.mjs:1:3394)
    at ServerDatabaseBackend.update (file:///home/ubuntu/app/resources/app/common/abstract/backend.mjs:153:31)
    at runMicrotasks (<anonymous>)
    at runNextTicks (internal/process/task_queues.js:58:5)
    at processImmediate (internal/timers.js:434:9)
    at async Socket.handleEvent (file:///home/ubuntu/app/resources/app/dist/server/sockets.mjs:1:568)
```

Also there were several repeated logs of

```
Compacted database file for combats
```

I was also using the [turn marker module](https://foundryvtt.com/packages/turnmarker), which I initially thought to be the culprit, but after restarting the server, and game, with this module disabled, the loop continued.  

Disabling this module allowed the game to be continued.

I was running version 0.8.9 of foundry and the latest version (1.3.1?) of this module.

**Comment by vtt-lair** [comment]
Thanks. I experienced the same in my own game but when I tested it again, I couldn't replicate it again. Still trying to figure out why it's happening.

**Comment by strongpauly** [comment]
I saw the same.  Ran through the combat with the same combatants and it was fine.  Tried to recreate it in a similar world but couldn't.  All I can add is that after restarting the first time it was fine until I progressed the turn order to the next combatant (which I think was part of a group), which then caused the "compacted database file" logs again, so I restarted quickly.  

It initially started when I added new combatants to a running combat (comprising of 1 new group of 20 combatants, 2 groups of 2 combatants (which didn't appear grouped), and a single combatant). Hope this helps!
